src/function/utility/count_if.cpp

```diff
@@ -0,0 +1,79 @@
+#include "function/scalar_function.h"
+#include "function/utility/scalar_utility_functions.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace function {
+
+struct CountIf {
+    template<class T>
+    static inline void operation(T& input, int128_t& result) {
+        if (input != 0) {
+            result = 1;
+        } else {
+            result = 0;
+        }
+    }
+};
+
+static void getExecFunc(LogicalTypeID operandTypeID, scalar_func_exec_t& func) {
+    switch (operandTypeID) {
+    case LogicalTypeID::SERIAL:
+    case LogicalTypeID::INT64: {
+        func = ScalarFunction::UnaryExecFunction<int64_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::INT32: {
+        func = ScalarFunction::UnaryExecFunction<int32_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::INT16: {
+        func = ScalarFunction::UnaryExecFunction<int16_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::INT8: {
+        func = ScalarFunction::UnaryExecFunction<int8_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::UINT64: {
+        func = ScalarFunction::UnaryExecFunction<uint64_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::UINT32: {
+        func = ScalarFunction::UnaryExecFunction<uint32_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::UINT16: {
+        func = ScalarFunction::UnaryExecFunction<uint16_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::UINT8: {
+        func = ScalarFunction::UnaryExecFunction<uint8_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::INT128: {
+        func = ScalarFunction::UnaryExecFunction<int128_t, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::DOUBLE: {
+        func = ScalarFunction::UnaryExecFunction<double, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::FLOAT: {
+        func = ScalarFunction::UnaryExecFunction<float, int128_t, CountIf>;
+    } break;
+    case LogicalTypeID::BOOL: {
+        func = ScalarFunction::UnaryExecFunction<bool, int128_t, CountIf>;
+    } break;
+    default:
+        KU_UNREACHABLE;
+    }
+}
+
+function_set CountIfFunction::getFunctionSet() {
+    function_set functionSet;
+    scalar_func_exec_t execFunc;
+    for (auto operandTypeID : LogicalTypeUtils::getNumericalLogicalTypeIDs()) {
+        getExecFunc(operandTypeID, execFunc);
+        functionSet.push_back(std::make_unique<ScalarFunction>(name,
+            std::vector<LogicalTypeID>{operandTypeID}, LogicalTypeID::INT128, execFunc));
+    }
+    getExecFunc(LogicalTypeID::BOOL, execFunc);
+    functionSet.push_back(std::make_unique<ScalarFunction>(name,
+        std::vector<LogicalTypeID>{LogicalTypeID::BOOL}, LogicalTypeID::INT128, execFunc));
```

Comment:
i think we can make the result type as UINT8, since the output is always 1 or 0.